chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/PendingSurveillanceManagerImpl.java

```diff
@@ -0,0 +1,728 @@
+            LOGGER.error(
+                    "Could not format " + surveillanceThresholdToProcessAsJobStr + " as an integer. Defaulting to"
+                            + " 50 instead.");
+        }
+        return surveillanceThresholdToProcessAsJob;
+    }
+
+    private void validate(final Surveillance surveillance) {
+        validator.validate(surveillance, false);
+    }
+
+    private void checkSurveillanceAuthority(final Surveillance surv) throws SurveillanceAuthorityAccessDeniedException {
+        Boolean hasOncAdmin = Util.isUserRoleAdmin();
+        Boolean hasAcbAdmin = Util.isUserRoleAcbAdmin();
+        if (StringUtils.isEmpty(surv.getAuthority())) {
+            // If user has ROLE_ADMIN and ROLE_ACB
+            // return 403
+            if (hasOncAdmin && hasAcbAdmin) {
+                String errorMsg = "Surveillance cannot be created by user having " + Authority.ROLE_ADMIN + " and "
+                        + Authority.ROLE_ACB;
+                LOGGER.error(errorMsg);
+                throw new SurveillanceAuthorityAccessDeniedException(errorMsg);
+            }
+        } else {
+            // Cannot have surveillance authority as ROLE_ADMIN for user lacking
+            // ROLE_ADMIN
+            if (surv.getAuthority().equalsIgnoreCase(Authority.ROLE_ADMIN) && !hasOncAdmin) {
+                String errorMsg = "User must have authority " + Authority.ROLE_ADMIN;
+                LOGGER.error(errorMsg);
+                throw new SurveillanceAuthorityAccessDeniedException(errorMsg);
+            } else if (surv.getAuthority().equalsIgnoreCase(Authority.ROLE_ACB)) {
+                // Cannot have surveillance authority as ACB for user lacking ONC
+                // and ACB roles
+                if (!hasOncAdmin && !hasAcbAdmin) {
+                    String errorMsg = "User must have ONC or ACB roles for a surveillance authority created by ACB";
+                    LOGGER.error(errorMsg);
+                    throw new SurveillanceAuthorityAccessDeniedException(errorMsg);
+                }
+            }
+        }
+    }
+
+    private Long createSurveillance(final Surveillance surv)
+            throws UserPermissionRetrievalException, SurveillanceAuthorityAccessDeniedException {
+        Long insertedId = null;
+        checkSurveillanceAuthority(surv);
+        updateNullAuthority(surv);
+
+        try {
+            insertedId = survDao.insertSurveillance(surv);
+        } catch (final UserPermissionRetrievalException ex) {
+            LOGGER.error("Error inserting surveillance.", ex);
+            throw ex;
+        }
+
+        return insertedId;
+    }
+
+    private void updateNullAuthority(final Surveillance surv) {
+        Boolean hasOncAdmin = Util.isUserRoleAdmin();
+        Boolean hasAcbAdmin = Util.isUserRoleAcbAdmin();
+        if (StringUtils.isEmpty(surv.getAuthority())) {
+            if (hasOncAdmin) {
+                surv.setAuthority(Authority.ROLE_ADMIN);
```

Comment:
I thought we were making all of the authorities either ROLE_ACB or ROLE_ONC. Or did I miss that discussion?

Reply:
Andrew I remember talking with you in a side chat when I was working on adding ROLE_ONC to begin with - we said the ADMIN authority should be renamed to ONC authority. I'm not sure if that discussion made it's way into the ticket. We should probably rename the authority all the way down into the database (migrate existing ones that are authority ADMIN to authority ONC - although I looked a couple weeks ago and there were none with ADMIN authority) and rename it in the code as well.

Reply:
Oh, we did talk about this, and recorded the answer in the ticket description:

> 2) When ADMIN uploads a surveillance, what authority is it? Answer: It is ROLE_ONC, at least for now. There is no ROLE_ADMIN authority for surveillance

Reply:
This value is overwritten by the DAO.  The function in the DAO is correct.  The method has been removed.